### PR TITLE
Providing more current examples of presentational elements in textboxes

### LINKED ISSUE #1493: Consider adding more normative requirements for authors on the textbox role
The ARIA spec says the following for the [`textbox` role](https://w3c.github.io/aria/#textbox):

> A type of input that allows free-form text as its value.
> 
> If the `aria-multiline` attribute is `true`, the widget accepts line breaks within the input, as in an HTML `textarea`. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input element, or cases in which an element with different semantics is repurposed as a text field.

After that there is a (non-normative) note and the characteristics table. That's it.  That suggests the following is, officially, a valid "simple text box":

```
<div role="textbox" aria-label="good luck!">
  <button>ok</button>
  <a href="#">link</a>
  <label>name: <input></label>
</div>
```

Some observations:
1. I don't think that's a "simple text box"
2. `contenteditable` isn't set but the default/implicit value of `aria-readonly` is `false`. According to the ARIA spec's [State and Property Attribute Processing](https://w3c.github.io/aria/#state_property_processing):

    > When WAI-ARIA roles are used, supported states and properties that are not present in the DOM are treated according to their default value.

    So we have to trust that the author is using some non-contenteditable means to make that `div` editable.

3. `aria-readonly` is `false` as described above. `aria-disabled` is also `false` for the same reason, but this `textbox` is also not focusable, which brings me to the next item:
4. Should this really be exposed as a textbox to ATs?

I'm currently working on fixing a Chromium bug related to authoring like the above which breaks the screen-reader user experience because those focusable children get pruned from the accessibility tree unless `contenteditable` is set on the `div` with the `textbox` role. Why are they getting pruned? Simple text boxes don't have children. (Right??)

Fixing the not-in-the-accessibility-tree bug is not hard. But I think authors need to be stopped from doing the above. And depending on the normative statements we add for authors, we might want to consider handling-author-errors content so that user agents can provide a more consistent implementation and user experience given such authoring.

### PR DESCRIPTION
**Pull request by chlane**
Closes #1493. Provided more current examples of presentational elements in textboxes that were suggested in  Pull Request https://github.com/w3c/aria/pull/1788.

***
<a href="https://pr-preview.s3.amazonaws.com/chlane/aria/pull/1830.html" title="Last updated on Feb 22, 2024, 8:14 PM UTC (75c884d)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria/1830/e88317c...chlane:75c884d.html" title="Last updated on Feb 22, 2024, 8:14 PM UTC (75c884d)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -9257,6 +9257,7 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to presentational elements such as icons used to visually convey information. Some examples are error and success icons, portraits or silhouettes of users and status indicators.</p>
```
[the remark this answers is not in the document]
Comment (cookiecrook) [reply index.html:9260]: I think "non-interactive" is better here than "presentational" but I could be convinced the original term is better.

**Review by cookiecrook — CHANGES_REQUESTED** [review]

File: index.html [hunk 2]
```diff
@@ -9332,7 +9333,7 @@ <h5>Presentational Role Inheritance</h5>
 					</tr>
 					<tr>
 						<th class="role-childpresentational-head" scope="row">Children Presentational:</th>
-						<td class="role-childpresentational"> </td>
+						<td class="role-childpresentational"> </td>
```
[the remark this answers is not in the document]
Comment (cookiecrook) [reply index.html:9336]: rolling back the whitespace diff

**Comment by cookiecrook** [comment]
I think your separate commit marked the rest of my suggestions as outdated. Please look back at the rest of the diff.

For future ref, you can commit reviewers diffs directly if you agree with the changes.

**Review by cookiecrook — CHANGES_REQUESTED** [review]
Most changes from the last review still unaddressed.

File: index.html (hunk 2 above)
Comment (cookiecrook) [inline index.html:9336]: I'm not able to roll back this diff with a change suggestion. Not sure why, but it's not as simple as a different whitespace char.

File: index.html [hunk 3]
```diff
@@ -9257,6 +9257,7 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive elements such as icons used to visually convey information. Some examples are error and success icons, portraits or silhouettes of users and status indicators.</p>
```
Comment (cookiecrook) [inline index.html:9260]:
```suggestion
				<p>Authors MUST limit the children of a textbox to non-interactive elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
				<ul>
					<li>an error icon, with an associated <sref>aria-invalid</sref> and/or <sref>aria-errormessage</sref>;</li>
					<li>an icon of a user silhouette, where the textbox is labeled as a name or username); and</li>
					<li>status indicators, such as characters remaining, which is exposed in text outside the field.</li>
					</p>
```

**Review by cookiecrook — CHANGES_REQUESTED** [review]
Adding back in the original diff.

**Review by scottaohara — COMMENTED** [review]
Does this PR also need to call out how it could potentially impact what is being defined in #1827?

For instance, unless explicitly called out as being content that is entirely presentational - could someone not do:
```
<div role=textbox ...>
  actual value 
  <img src=... alt="oh no, now am i'm part of the value too!?"> <!-- error icon that really should probably have been positioned outside of the textbox anyway -->
</div>
```

**Comment by chlane** [comment]
@scottaohara @cookiecrook I think the example Scott provided below favors the usage of "entirely presentational" instead of "non-interactive". Or could we say "entirely presentational and non-interactive"?

```
<div role=textbox ...>
  actual value 
  <img src=... alt="oh no, now am i'm part of the value too!?"> 
  <!-- error icon that really should probably have been positioned outside of the textbox anyway -->
</div>
```

File: index.html [hunk 4]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
```
Comment (cookiecrook) [inline index.html:9260]:
```suggestion
				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
```

**Review by cookiecrook — APPROVED** [review]

File: index.html (hunk 4 above)
Comment (cookiecrook) [reply index.html:9260]: Agree with Scott and Chris on the image alt example.

File: index.html [hunk 5]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
+					<ul>
+						<li>an error icon, with an associated <sref>aria-invalid</sref> and/or <sref>aria-errormessage</sref>;</li>
+						<li>an icon of a user silhouette, where the textbox is labeled as a name or username; and</li>
```
Comment (scottaohara) [inline index.html:9263]:
```suggestion
						<li>an icon of a user silhouette, where the textbox is also visibly labeled or provided an accessible name of "name" or "username"; and</li>
```

File: index.html [hunk 6]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
+					<ul>
+						<li>an error icon, with an associated <sref>aria-invalid</sref> and/or <sref>aria-errormessage</sref>;</li>
```
Comment (scottaohara) [inline index.html:9262]:
```suggestion
						<li>an error icon, where the containing textbox has been provided an <sref>aria-invalid</sref>,  <sref>aria-errormessage</sref>, or both attributes;</li>
```

File: index.html [hunk 7]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
...
+						<li>an icon of a user silhouette, where the textbox is labeled as a name or username; and</li>
+						<li>status indicators, such as characters remaining, which is exposed in text outside the field.</li>
```
Comment (scottaohara) [inline index.html:9264]:
```suggestion
						<li>a graphical status indicator, such as a gauge to represent characters remaining, which represents dynamically updating text available outside of the textbox.</li>
```

**Review by scottaohara — COMMENTED** [review]
just some final wording suggestions, then i think this would be good to go.

**Comment by chlane** [comment]
@scottaohara thanks for providing more detail where needed. I've commit all the suggestions.

File: index.html [hunk 8]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
...
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
+					<ul>
+						<li>an error icon, where the containing textbox has been provided an <sref>aria-invalid</sref>,  <sref>aria-errormessage</sref>, or both attributes;</li>
+						<li>an icon of a user silhouette, where the textbox is also visibly labeled or provided an accessible name of "name" or "username"; and</li>
+						<li>a graphical status indicator, such as a gauge to represent characters remaining, which represents dynamically updating text available outside of the textbox.</li>
+					</ul>
+				</p>
```
Comment (scottaohara) [inline index.html:9266]: remove stray ending paragraph tag.
```suggestion
```

File: index.html [hunk 9]
```diff
@@ -9257,6 +9257,13 @@ <h5>Presentational Role Inheritance</h5>
 			<div class="role-description">
 				<p>A type of input that allows free-form text as its value.</p>
 				<p>If the <pref>aria-multiline</pref> <a>attribute</a> is <code>true</code>, the <a>widget</a> accepts line breaks within the input, as in an <abbr title="Hypertext Markup Language">HTML</abbr> <code>textarea</code>. Otherwise, this is a simple text box. The intended use is for languages that do not have a text input <a>element</a>, or cases in which an element with different <a>semantics</a> is repurposed as a text field.</p>
+				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:
```
Comment (scottaohara) [inline index.html:9260]: close paragraph.  a list cannot be nested within a paragraph.
```suggestion
				<p>Authors MUST limit the children of a textbox to non-interactive, entirely presentational elements such as icons used to visually convey information that is already exposed in an accessible manner. Examples include:</p>
```

**Review by scottaohara — APPROVED** [review]
some markup cleanup and approving this with the changes made

File: index.html (hunk 2 above)
Comment (spectranaut) [reply index.html:9341]: @chlane do you think you can remove this change? other wise this looks good to go!

**Review by spectranaut — APPROVED** [review]

File: index.html (hunk 2 above)
Comment (chlane) [reply index.html:9346]: @spectranaut not sure why this whitespace issue is happening. I couldn't tell what to change on my end but the commit button was enabled after I edited line 9345. I just deleted and replaced the whitespace.

**Pull request merged**